x-pack/plugins/discover_log_explorer/common/datasets/v1/find_integrations.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { HttpFetchQuery } from '@kbn/core-http-browser';
+import * as rt from 'io-ts';
+import { integrationRT } from '../types';
+import { formatSearch, sortOrderRT } from './common';
+
+const searchAfterRT = rt.array(rt.union([rt.number, rt.string]));
+
+export const findIntegrationsResponseRT = rt.exact(
+  rt.intersection([
+    rt.type({
+      items: rt.array(integrationRT),
+      total: rt.number,
+    }),
+    rt.partial({
+      searchAfter: searchAfterRT,
+    }),
+  ])
+);
+
+export const findIntegrationsRequestQueryRT = rt.exact(
+  rt.partial({
+    nameQuery: rt.string,
+    perPage: rt.number,
+    dataStreamType: rt.literal('logs'),
+    searchAfter: searchAfterRT,
```

Comment:
I just realized we don't even need the intermediate type I proposed earlier to encode the request params. This should replace the `formatSearch` logic pretty elegantly:

```suggestion
    searchAfter: jsonRt.pipe(searchAfterRT),
```

What do you think of that?

Reply:
Short and simple, I love it! I have to take care of exclude the property from the search if `searchAfter === undefined`, but that should be fast

Reply:
I think the `http` service might even remove the query params with `undefined` values itself: https://github.com/tonyghiani/kibana/blob/51db3a13becdd930847100e90a97b867a09ad65f/packages/core/http/core-http-browser-internal/src/fetch.ts#L146